vite.config.ts

```diff
@@ -1,25 +1,65 @@
 /// <reference types="vitest" />
 import react from "@vitejs/plugin-react"
+import { consola } from "consola"
+import dotenv from "dotenv"
 import path, { resolve } from "path"
-import { defineConfig } from "vite"
+import { defineConfig, Plugin } from "vite"
 import dts from "vite-plugin-dts"
 import { libInjectCss } from "vite-plugin-lib-inject-css"
+import { buildSyncPlugin } from "./vite/build-sync.plugin"
+
+dotenv.config({
+  path: [".env.local", ".env"],
+})
+
+const extraPlugins: Plugin[] = []
+
+/* Build sync */
+const buildSyncArg = process.argv.find((arg) => arg.startsWith("--buildSync"))
+const buildSync = !!buildSyncArg
+const buildSyncValue = buildSyncArg
+  ? buildSyncArg.split("=")[1] || process.env.CODER_REMOTE
+  : null
+
+if (buildSync) {
+  if (!buildSyncValue) {
+    consola.error(
+      "The buildSync flag must remote target or you can set it in the env variable CODER_REMOTE in the `.env.local` file"
+    )
+    process.exit(1)
+  }
+
+  const [remote, remoteFolder] = buildSyncValue.split(":")
+
+  const target = [
+    remote,
+    remoteFolder ||
+      "/home/factorial/workspace/factorial/frontend/node_modules/@factorialco/factorial-one",
```

Comment:
maybe name it for better understanding where we point to if the variable is not set

Reply:
done!